flexget/task.py

```diff
@@ -428,8 +434,27 @@ def find_entry(self, category='entries', **values):
                 return entry
         return None
 
-    def plugins(self, phase=None):
-        """Get currently enabled plugins.
+    def get_plugin_by_name(self, name: str) -> PluginInfo:
+        if name in self.disabled_plugins:
+            raise ValueError(f'`{name}` plugin is disabled for this task.')
```

Comment:
I think a dedicated `DisabledPluginError` would be more explicit and obvious in this case

Reply:
Yeah, that might be better.